app/main.py

```diff
@@ -0,0 +1,362 @@
+    gis_dataframe: pd.DataFrame
+        Dataframe containing GIS data - the dataframe to enrich.
+    mrid_dataframe: pd.DataFrame
+        Dataframe containing ETS data
+    gis_line_name_to_ets_line_name: dict[str, str]
+        Mapping between gis line name and ets line name
+    ets_name_to_mrid: dict
+        Mapping between ets line name and mrid
+    bad_line_names: list[str]
+        List of gis line names that should be removed from the dataset
+        (i.e. lines that could not be translated into ETS line names).
+    gis_column_name: str
+        Name of the column in gis_dataframe which contains the gis line names.
+    new_column_name: str
+        Name of the (new) column that should store the ets line names
+    ets_name_column: str
+        Name of the column in mrid_dataframe that contains the ets names.
+    ets_dlr_enabled_column: str
+        Name of the column in mrid_dataframe which contains enabled state
+
+    Returns
+    -------
+    pd.DataFrame
+        Clone of gis_dataframe enriched with MRID data, minus non-translated rows.
+    '''
+    # Checking to see if there are lines in the MRID list with enabled DLR that can not be found in GIS data
+    lines_with_dlr_enabled = mrid_dataframe[mrid_dataframe[ets_dlr_enabled_column].str.upper() == "YES"][ets_column_name]
+    missing_gis_data = list(set(lines_with_dlr_enabled) - set(gis_line_name_to_ets_line_name.values()))
+
+    if missing_gis_data:
+        log.error(f'Some lines set up for DLR have no GIS data available, they are:{missing_gis_data}')
+    else:
+        log.info('Gis data is found for all lines in the MRID file with DLR enabled.')
+
+    # Generate a dictonary map used to enrich the gis_dataframe with MRID
+    gis_name_to_mrid = {k: ets_name_to_mrid.get(v, v) for k, v in gis_line_name_to_ets_line_name.items()}
+    gis_dataframe[new_column_name] = gis_dataframe[gis_column_name].map(gis_name_to_mrid)
+
+    # Removing any names that could not be translated to represent a clean dataframe that only contains
+    # information that can be looked up with MRID.
+    gis_dataframe = gis_dataframe[~gis_dataframe[gis_column_name].isin(bad_line_names)]
+
+    return gis_dataframe
+
+
+def verify_translated_names_againts_ets(translate_gis_name_to_ets: dict[str, str], mrid_dataframe: pd.DataFrame,
+                                        ets_column_name: str):
+    '''
+    The function checks if there are lines in the ETS dataset which is not represented in the GIS dataset.
+    This is for logging and to be able to see if with a new dataset the comparing is getting a lesser quality.
+
+    Parameters
+    ----------
+    gis_line_name_to_ets_line_name: dict
+        The dict containing the translated names from GIS.
+    mrid_dataframe: pd.DataFrame
+        Dataframe with all ETS information in.
+    ETS_COLUMN_NAME: str
+        The specific column in the MRID dataframe that contains the names of lines.
+
+    Returns
+    -------
+        None
```

Comment:
```suggestion
    None
```
I think the correct procedure when there is no return is actually to just skip the Returns section - but if you leave it, there is no indentation under headers.